src/main/kotlin/com/makarytskyi/rentcar/service/impl/OrderServiceImpl.kt

```diff
@@ -22,69 +28,81 @@ internal class OrderServiceImpl(
     private val userRepository: UserRepository,
 ) : OrderService {
 
-    override fun getById(id: String): AggregatedOrderResponse =
-        orderRepository.findById(id)?.let { AggregatedOrderResponse.from(it) }
-            ?: throw NotFoundException("Order with id $id is not found")
-
-    override fun findAll(page: Int, size: Int): List<AggregatedOrderResponse> =
-        orderRepository.findAll(page, size).map { AggregatedOrderResponse.from(it) }
-            .toList()
-
-    override fun create(createOrderRequest: CreateOrderRequest): OrderResponse {
-        validateDates(createOrderRequest.from, createOrderRequest.to)
-        validateUserExists(createOrderRequest.userId)
-        validateCarAvailability(createOrderRequest.carId, createOrderRequest.from, createOrderRequest.to)
-
-        return OrderResponse.from(
-            orderRepository.create(CreateOrderRequest.toEntity(createOrderRequest)),
-            getCarPrice(createOrderRequest.carId)
-        )
+    override fun getById(id: String): Mono<AggregatedOrderResponse> =
+        orderRepository.findFullById(id)
+            .switchIfEmpty { Mono.error(NotFoundException("Order with id $id is not found")) }
+            .map { AggregatedOrderResponse.from(it) }
+
+    override fun findAll(page: Int, size: Int): Flux<AggregatedOrderResponse> =
+        orderRepository.findFullAll(page, size).map { AggregatedOrderResponse.from(it) }
+
+    override fun create(createOrderRequest: CreateOrderRequest): Mono<OrderResponse> {
+        return createOrderRequest.toMono()
+            .doOnNext { validateDates(it.from, it.to) }
+            .flatMap {
+                Mono.zip(
+                    Mono.defer { validateUserExists(createOrderRequest.userId) }.subscribeOn(Schedulers.parallel()),
+                    Mono.defer {
+                        validateCarAvailability(
+                            createOrderRequest.carId,
+                            createOrderRequest.from,
+                            createOrderRequest.to
+                        )
+                    }.subscribeOn(Schedulers.parallel())
+                ).thenReturn(createOrderRequest)
```

Comment:
do you need to return request here or you can return just Unit?

Reply:
I can return just Unit.